actors/builtin/miner/miner_test.go

```diff
@@ -728,6 +732,79 @@ func TestWindowPost(t *testing.T) {
 			actor.submitWindowPoSt(rt, deadline, partitions, infos, cfg)
 		})
 	})
+
+	t.Run("detects faults from previous missed posts", func(t *testing.T) {
+		rt := builder.Build(t)
+
+		// skip two PoSts
+		_, infos1, _ := runTillFirstDeadline(rt)
```

Comment:
I'm mildly worried that we aren't fully testing what we want.  The reason is that I think infos1 will always be empty because `handleProvingPeriod`'s `AssignNewSectors` call always skips deadline 0 as being unassignable.  If they are empty than we always get 0 contribution to fee whether or not we implement late/non-late split correctly.

I could be reading the deadline math wrong in `advancePastProvingPeriodWithCron` but I don't think it skips deadline 0 for us.

Reply:
`runTillFirstDeadline` calls `runTillNextDeadline` at the end to skip past the zero deadline. The concern is valid though. I've added some assertions to ensure these deadlines have sectors even if the logic changes.